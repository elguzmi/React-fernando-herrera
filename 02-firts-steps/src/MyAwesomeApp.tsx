import type{ CSSProperties } from "react";

export function MyAwesomeApp() {
    const firtsName = "Santiago" , lastName = 'Guzman'

    const favoriteGames = ['elden ring' , 'Halo', 'The last of us']
    const isActive = true;

    const address = {
        zipCode : '123',
        country : 'Colombia'
    }

    const styles:CSSProperties = {
        backgroundColor: isActive ? '#fafa' : '#ccc',
        borderRadius: 10,
        padding: 10
    }
    return (
        <div data-testid="div-app">
            <h1 data-testid="firts-name-title"> { firtsName } </h1>
            <h3 data-testid="last-name-title">{ lastName }</h3>

            <p className="favorite-games">
                { favoriteGames.join(', ') }
            </p>

            <h1 className="active">{ isActive && 'Activo' }</h1>
            <p>{ 2 + 2}</p>

            <p style={styles}>
                { JSON.stringify(address)  }
            </p>
        </div>
    )
}