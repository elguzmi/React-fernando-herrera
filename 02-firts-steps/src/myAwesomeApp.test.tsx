import {describe, expect, test} from 'vitest'
import { MyAwesomeApp } from './MyAwesomeApp'
import { render , screen ,} from '@testing-library/react'

describe('myAwesomeApp',()=>{
  test('should render firts name and lastname' , ()=>{
    // Crea como un dom virtual
    const {container} = render(<MyAwesomeApp />); // el container no se actualiza con eventos por eso se utiliza screen
    //el componente renderizado en el DOM
    //console.log(container.innerHTML);

    const textH1 = container.querySelector('h1'); // regresa el primer h1 que encuentre
    expect(textH1?.innerHTML).toContain('Santiago');

   // Si falla una prueba no se va a ejecutar el siguiente test

    const textH3 = container.querySelector('h3'); 
    expect(textH3?.innerHTML).toContain('Guzman');
  })

  test('should render the firts name and lastname - Screen',()=>{
    render(<MyAwesomeApp />);
    screen.debug();
    //const textH1 = screen.getByRole('heading',{level:1});
    const textH1 = screen.getByTestId('firts-name-title');
    expect(textH1.innerHTML).toContain('Santiago');

    const textH3 = screen.getByRole('heading',{level:3});
    expect(textH3.innerHTML).toContain('Guzman');

  })
  

  test('should match with snapshot',()=>{
    // el snapshot es un caracteristica de vitest
    const {container} = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot();
  })

  test('should match snapshot with screen',()=>{
    render(<MyAwesomeApp />);
    expect(screen.getByTestId('div-app')).toMatchSnapshot();
  })

})