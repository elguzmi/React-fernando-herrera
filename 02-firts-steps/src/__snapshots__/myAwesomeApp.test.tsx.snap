// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`myAwesomeApp > should match snapshot with screen 1`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="firts-name-title"
  >
     
    Santiago
     
  </h1>
  <h3
    data-testid="last-name-title"
  >
    Guzman
  </h3>
  <p
    class="favorite-games"
  >
    elden ring, Halo, The last of us
  </p>
  <h1
    class="active"
  >
    Activo
  </h1>
  <p>
    4
  </p>
  <p
    style="background-color: rgba(255, 170, 255, 0.667); border-radius: 10px; padding: 10px;"
  >
    {"zipCode":"123","country":"Colombia"}
  </p>
</div>
`;

exports[`myAwesomeApp > should match with snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="firts-name-title"
    >
       
      Santiago
       
    </h1>
    <h3
      data-testid="last-name-title"
    >
      Guzman
    </h3>
    <p
      class="favorite-games"
    >
      elden ring, Halo, The last of us
    </p>
    <h1
      class="active"
    >
      Activo
    </h1>
    <p>
      4
    </p>
    <p
      style="background-color: rgba(255, 170, 255, 0.667); border-radius: 10px; padding: 10px;"
    >
      {"zipCode":"123","country":"Colombia"}
    </p>
  </div>
</div>
`;
